README
This is what I consider to be a fairly feature-filled home decor mod.  
It supplies a bunch of stuff found in most homes, like flower pots, home 
electronics, brass and wrought-iron tables, and lots more!

All items can be accessed either by crafting various other items 
together, or with the usual /give commands.  To get a list of the actual 
node names, just go into the homedecor/ folder and run the listnodes.sh 
Bash script.  Note that a few of the listed nodes are kinda useless by 
themselves (like the various parts of the folding doors).

All of the images used for the recipes below are included in the 
crafting-guide/ folder.

Most stuff can be made from materials found through a game world.  Some 
stuff can only be crafted by using materials from another mod (look for 
the orange highlights).  Currently, this mod can use materials from 
moreores, mesecons, wool, and unifieddyes if present. You can still use 
the usual /give commands to get the items that depend on these mods if 
you don't use them installed.

This mod is still a work-in-progress, but should be complete enough not 
to irritate anyone. :-) Namely, dressers still need to be fleshed out, 
many items can be used as fuel but the burn times need tuning, and I 
need to tune the cook time and fuel usage on the four smelted items.

This mod was put together using a recent git pull, and requires a 
build/pull dated June 17, 2012 or later.  Most stuff should work fine on 
older builds as well, but a few nodes in this mod require Nodebox 
support, a feature added to Minetest on that date.  It will NOT work 
with Minetest v0.3 (that version doesn't support mods).

Much of the code (especially for the doors) plus the bucket were taken 
from the files which come with Minetest and altered to suit.  Many 
thanks to Jeija for coming up with an elegant cylinder using nodeboxes; 
see his 'irregular' mods).  His code is used herein (you can guess where 
:-) ).

Dependencies: none (just the game's default stuff)

Recommends: buckets, flowers, unifieddyes, junglegrass, moreores, 
mesecons, wool

License: GPL for all lua code.  WTFPL for all of my textures and 
everything else.

The distribution archive also supplies a copy of Ironzorg's Flowers mod 
and of my Unified Dyes mod, since this mod provides more stuff when 
those two are present.  You can remove these if you want - they're only 
needed if you want to craft the various dyed/colored items.


changelog.txt
Changelog
---------
2012-09-25: Added alternate recipes for most items that use dyes.  Removed 
Mesecons dependency from TV and stereo (uses in-built silicon lumps if Mesecons 
isn't present).  Replaced speakers' Mesecons Noteblocks with copper ingots from 
Moreores.

2012-08-01: Added optional recipes for items that use wool to also use cotton
instead.  Choose one or the other in any given recipe (you can't mix the two).
No, it isn't a Kosher thing. ;-)

2012-07-28: Noticed I technically had left/right textures swapped on a few 
blocks.  The textures were drawn to counteract this, without my realizing I was 
doing just that. :-) Fixed.  Also replaced all "tile_images" references with 
"tiles", since the former is deprecated in favor of the latter.

2012-07-26:  Updated to use the Vessels mod to give back empty dye bottles.

2012-07-24: Made all objects that use a dye require a more appropriate color, 
made all objects that use a dye give back empty bottles on craft, made grey 
shutters use the various grey paints directly.  Moved this changelog out of the 
forum post and into this separate file.

2012-07-12: Moved project to github.

2012-07-02: Deleted a few redundant files from the unified dyes mod.

2012-06-30: Added a fancy wood-and-glass door.  Textures used here were 
formerly used for (and have been replaced by new textures for) the default wood 
door in my realistic texture packs.

2012-06-27: Relaxed dependency on UnifiedDyes to prevent depending on Flowers.  
The mod will still use them if available; if not, you just can't craft the 
various colored objects.

2012-06-26: Separated dyes into their own mod ("unifieddyes").  Added a copy of 
flowers and made this mod depend on it and unified dyes.  Updated all crafting 
recipes to fit, got rid of the conditional code for flowers (since it is a 
dependency now), removed a bunch of redundant register_craftitem code, fixed 
some copy&paste errors on the window shutters.  As a consequence of these 
changes, white paint, titanium dioxide, and all colors of dyes will have to be 
removed and replaced (but the things that depend on them are fine).

2012-06-25: Removed all of the old legacy_wallmounted references - I didn't 
realize they were deprecated (and they were causing out-of-nodes issues also).

2012-06-23: Added small speaker, round brass pole, square wrought iron pole 
(without connecting rungs).  Fairly recent version of Minetest required for 
these to work (uses nodeboxes).  Note that if you've placed brass or wrought 
iron poles in your world already, they will be drawn using these newer styles; 
if you need the square poles and/or connecting rungs, you'll need to remove 
them and craft brass/wrought iron fences instead.

2012-06-16: Tweaked terracotta roof tiles/shingles to give the 4 pieces back on 
digging a 'shingle'.

2012-06-15 (a bit later): Did the same for the stereo and television and 
improved them in the process.

2012-06-15: Tweaked the speaker textures, added separate images for all 6 
sides.

2012-06-14: Added right-opening versions of oak and mahogany "folding" doors - 
allows one to create double doors.  Split the two colors into separate files.  
Inspired by Calinou's update to the default doors mod - works the same way too. 
:-)

2012-06-13 (a lot later): Changed the crafting recipe slightly for nightstands 
- they're designated as 'oak' now.  Added crafting recipe for mahogany 
variants.  If you've already placed nightstands into your world, you should 
remove them before updating - sorry, you'll have to rebuild them due my having 
had to change the node names.  Added initial crafting recipes for oak and 
mahogany drawers and complete dressers (dresser components are not yet useful, 
work in progress.  See crafts.lua for more details).

2012-06-13: Changed buckets of white paint so that the empty bucket is returned 
when the bucket of paint is crafted into something else (rather than when the 
bucket of paint is first created).  * Fixed copy&paste error from the previous 
bucket change.  * Moved smelting and crafting of white paint from crafts.lua 
over to dyes.lua.

2012-06-12: Multiple post-release updates over the course of the day. * Fixed 
recipe collision with glass panes.  * Implemented a full set of dyes derived 
from flowers, updated all related crafting recipes and screenshots to match. * 
Collected all dyes into one file.  Changed recipes for blue and purple dyes to 
something that makes a bit more sense. :-) * Tweaked oak shutters to give two 
pieces on crafting. * Added alternate recipes for orange and green dyes.

2012-06-12: Initial release.


fences.lua
-- This file adds fences of various types

minetest.register_node("homedecor:fence_brass", {
        description = "Brass Fence/railing",
        drawtype = "fencelike",
        tiles = {"homedecor_tile_brass.png"},
        inventory_image = "homedecor_fence_brass.png",
        wield_image = "homedecor_pole_brass.png",
        paramtype = "light",
        is_ground_content = true,
        selection_box = {
                type = "fixed",
                fixed = {-1/7, -1/2, -1/7, 1/7, 1/2, 1/7},
        },
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
})

minetest.register_node("homedecor:fence_wrought_iron", {
        description = "Wrought Iron Fence/railing",
        drawtype = "fencelike",
        tiles = {"homedecor_tile_wrought_iron.png"},
        inventory_image = "homedecor_fence_wrought_iron.png",
        wield_image = "homedecor_pole_wrought_iron.png",
        paramtype = "light",
        is_ground_content = true,
        selection_box = {
                type = "fixed",
                fixed = {-1/7, -1/2, -1/7, 1/7, 1/2, 1/7},
        },
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
})

minetest.register_node("homedecor:fence_picket", {
	drawtype = "nodebox",
        description = "Unpainted Picket Fence",
        tiles = {
		"homedecor_fence_picket.png",
		"homedecor_fence_picket.png",
		"homedecor_fence_picket.png",
		"homedecor_fence_picket.png",
		"homedecor_fence_picket_backside.png",
		"homedecor_fence_picket.png"
	},
--        inventory_image = "homedecor_fence_picket.png",
--        wield_image = "homedecor_fence_picket.png",
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.4, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.495, 0.5, 0.5, 0.495 }
        },
})

minetest.register_node("homedecor:fence_picket_white", {
	drawtype = "nodebox",
        description = "White Picket Fence",
        tiles = {
		"homedecor_fence_picket_white.png",
		"homedecor_fence_picket_white.png",
		"homedecor_fence_picket_white.png",
		"homedecor_fence_picket_white.png",
		"homedecor_fence_picket_white_backside.png",
		"homedecor_fence_picket_white.png"
	},
--        inventory_image = "homedecor_fence_picket_white.png",
--        wield_image = "homedecor_fence_picket_white.png",
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.4, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.495, 0.5, 0.5, 0.495 }
        },
})

minetest.register_node("homedecor:fence_privacy", {
	drawtype = "nodebox",
        description = "Wooden Privacy Fence",
        tiles = {
		"homedecor_fence_privacy_tb.png",
		"homedecor_fence_privacy_tb.png",
		"homedecor_fence_privacy_sides.png",
		"homedecor_fence_privacy_sides.png",
		"homedecor_fence_privacy_backside.png",
		"homedecor_fence_privacy_front.png"
	},
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 5/16, 0.5, 0.5, 8/16 }
        },
        node_box = {
                type = "fixed",
		fixed = {
			{ -8/16, -8/16, 5/16, -5/16,  8/16, 7/16 },	-- left part
			{ -4/16, -8/16, 5/16,  3/16,  8/16, 7/16 },	-- middle part
			{  4/16, -8/16, 5/16,  8/16,  8/16, 7/16 },	-- right part
			{ -8/16, -2/16, 7/16,  8/16,  2/16, 8/16 },	-- connecting rung
		}
        },
})

minetest.register_node("homedecor:fence_barbed_wire", {
	drawtype = "nodebox",
        description = "Barbed Wire Fence",
        tiles = {"homedecor_fence_barbed_wire.png"},
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.375, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
		fixed = {
			{ -8/16, -8/16, 6/16, -6/16, 8/16, 8/16 },	-- left post
			{  6/16, -8/16, 6/16,  8/16, 8/16, 8/16 }, 	-- right post
			{ -6/16, -8/16, 7/16,  6/16, 8/16, 7/16 }	-- the wire
		}		
        },
})

minetest.register_node("homedecor:fence_chainlink", {
	drawtype = "nodebox",
        description = "Chainlink Fence",
        tiles = {
		"homedecor_fence_chainlink_tb.png",
		"homedecor_fence_chainlink_tb.png",
		"homedecor_fence_chainlink_sides.png",
		"homedecor_fence_chainlink_sides.png",
		"homedecor_fence_chainlink_fb.png",
		"homedecor_fence_chainlink_fb.png",
	},
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.375, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
		fixed = {
			{ -8/16, -8/16,  6/16, -7/16,  8/16,  8/16 },	-- left post
			{  7/16, -8/16,  6/16,  8/16,  8/16,  8/16 }, 	-- right post
			{ -8/16,  7/16, 13/32,  8/16,  8/16, 15/32 },	-- top piece
			{ -8/16, -8/16, 13/32,  8/16, -7/16, 15/32 },	-- bottom piece
			{ -8/16, -8/16,  7/16,  8/16,  8/16,  7/16 }	-- the chainlink itself
		}		
        },
})


-- =====
-- Gates

minetest.register_node("homedecor:fence_picket_gate_closed", {
	drawtype = "nodebox",
        description = "Unpainted Picket Fence Gate",
        tiles = {
		"homedecor_fence_picket_gate.png",
		"homedecor_fence_picket_gate.png",
		"homedecor_fence_picket_gate.png",
		"homedecor_fence_picket_gate.png",
		"homedecor_fence_picket_gate_backside.png",
		"homedecor_fence_picket_gate.png"
	},
--        inventory_image = "homedecor_fence_picket_gate.png",
--        wield_image = "homedecor_fence_picket_gate.png",
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.4, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.495, 0.5, 0.5, 0.495 }
        },
})

minetest.register_node("homedecor:fence_picket_gate_open", {
	drawtype = "nodebox",
        description = "Unpainted Picket Fence Gate",
        tiles = {
		"homedecor_fence_picket_gate.png",
		"homedecor_fence_picket_gate.png",
		"homedecor_fence_picket_gate.png",
		"homedecor_fence_picket_gate_backside.png",
		"homedecor_fence_picket_gate.png",
		"homedecor_fence_picket_gate.png"
	},
--        inventory_image = "homedecor_fence_picket_gate.png",
--        wield_image = "homedecor_fence_picket_gate.png",
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3, not_in_creative_inventory=1},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { 0.4, -0.5, -0.5, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
                fixed = { 0.495, -0.5, -0.5, 0.495, 0.5, 0.5 }
        },
	drop = "homedecor:fence_picket_gate_closed"
})

minetest.register_node("homedecor:fence_picket_gate_white_closed", {
	drawtype = "nodebox",
        description = "White Picket Fence Gate",
        tiles = {
		"homedecor_fence_picket_gate_white.png",
		"homedecor_fence_picket_gate_white.png",
		"homedecor_fence_picket_gate_white.png",
		"homedecor_fence_picket_gate_white.png",
		"homedecor_fence_picket_gate_white_backside.png",
		"homedecor_fence_picket_gate_white.png"
	},
--        inventory_image = "homedecor_fence_picket_gate.png",
--        wield_image = "homedecor_fence_picket_gate.png",
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.4, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.495, 0.5, 0.5, 0.495 }
        },
})

minetest.register_node("homedecor:fence_picket_gate_white_open", {
	drawtype = "nodebox",
        description = "White Picket Fence Gate",
        tiles = {
		"homedecor_fence_picket_gate_white.png",
		"homedecor_fence_picket_gate_white.png",
		"homedecor_fence_picket_gate_white.png",
		"homedecor_fence_picket_gate_white_backside.png",
		"homedecor_fence_picket_gate_white.png",
		"homedecor_fence_picket_gate_white.png"
	},
--        inventory_image = "homedecor_fence_picket_gate.png",
--        wield_image = "homedecor_fence_picket_gate.png",
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3, not_in_creative_inventory=1},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { 0.4, -0.5, -0.5, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
                fixed = { 0.495, -0.5, -0.5, 0.495, 0.5, 0.5 }
        },
	drop = "homedecor:fence_picket_gate_closed"
})

minetest.register_node("homedecor:fence_barbed_wire_gate_closed", {
	drawtype = "nodebox",
        description = "Barbed Wire Fence Gate",
        tiles = {
		"homedecor_fence_barbed_wire_gate_edges.png",
		"homedecor_fence_barbed_wire_gate_edges.png",
		"homedecor_fence_barbed_wire_gate_edges.png",
		"homedecor_fence_barbed_wire_gate_edges.png",
		"homedecor_fence_barbed_wire_gate_backside.png",
		"homedecor_fence_barbed_wire_gate_front.png"
	},
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.375, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
		fixed = {
			{ -8/16, -8/16, 6/16, -6/16, 8/16, 8/16 },	-- left post
			{  6/16, -8/16, 6/16,  8/16, 8/16, 8/16 }, 	-- right post
			{ -8/16,  7/16, 13/32,  8/16,  8/16, 15/32 },	-- top piece
			{ -8/16, -8/16, 13/32,  8/16, -7/16, 15/32 },	-- bottom piece
			{ -6/16, -8/16, 7/16,  6/16, 8/16, 7/16 }	-- the wire
		}		
        },
})

minetest.register_node("homedecor:fence_barbed_wire_gate_open", {
	drawtype = "nodebox",
        description = "Barbed Wire Fence Gate",
        tiles = {
		"homedecor_fence_barbed_wire_gate_edges.png",
		"homedecor_fence_barbed_wire_gate_edges.png",
		"homedecor_fence_barbed_wire_gate_front.png",
		"homedecor_fence_barbed_wire_gate_backside.png",
		"homedecor_fence_barbed_wire_gate_edges.png",
		"homedecor_fence_barbed_wire_gate_edges.png"
	},
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3, not_in_creative_inventory=1},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { 0.375, -0.5, -0.5, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
		fixed = {
			{  6/16, -8/16, -8/16,  8/16,  8/16, -6/16 },	-- left post
			{  6/16, -8/16,  6/16,  8/16,  8/16,  8/16 }, 	-- right post
			{ 13/32,  7/16, -8/16, 15/32,  8/16,  8/16 },	-- top piece
			{ 13/32, -8/16, -8/16, 15/32, -7/16,  8/16 },	-- bottom piece
			{  7/16, -8/16, -6/16,  7/16,  8/16,  6/16 }	-- the wire
		}		
        },
	drop = "homedecor:fence_barbed_wire_gate_closed"
})

minetest.register_node("homedecor:fence_chainlink_gate_closed", {
	drawtype = "nodebox",
        description = "Chainlink Fence Gate",
        tiles = {
		"homedecor_fence_chainlink_gate_tb.png",
		"homedecor_fence_chainlink_gate_tb.png",
		"homedecor_fence_chainlink_gate_sides.png",
		"homedecor_fence_chainlink_gate_sides.png",
		"homedecor_fence_chainlink_gate_backside.png",
		"homedecor_fence_chainlink_gate_front.png",
	},
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { -0.5, -0.5, 0.375, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
		fixed = {
			{ -8/16, -8/16,  6/16, -7/16,  8/16,  8/16 },	-- left post
			{  6/16, -8/16,  6/16,  8/16,  8/16,  8/16 }, 	-- right post
			{ -8/16,  7/16, 13/32,  8/16,  8/16, 15/32 },	-- top piece
			{ -8/16, -8/16, 13/32,  8/16, -7/16, 15/32 },	-- bottom piece
			{ -8/16, -8/16,  7/16,  8/16,  8/16,  7/16 },	-- the chainlink itself
			{ -8/16, -3/16,  6/16, -6/16,  3/16,  8/16 }	-- the lump representing the lock
		}		
        },
})

minetest.register_node("homedecor:fence_chainlink_gate_open", {
	drawtype = "nodebox",
        description = "Chainlink Fence Gate (open)",
        tiles = {
		"homedecor_fence_chainlink_gate_tb.png",
		"homedecor_fence_chainlink_gate_tb.png",
		"homedecor_fence_chainlink_gate_front.png",
		"homedecor_fence_chainlink_gate_backside.png",
		"homedecor_fence_chainlink_gate_sides.png",
		"homedecor_fence_chainlink_gate_sides.png",
	},
        paramtype = "light",
        is_ground_content = true,
        groups = {snappy=3, not_in_creative_inventory=1},
        sounds = default.node_sound_wood_defaults(),
	walkable = true,
	paramtype2 = "facedir",
        selection_box = {
                type = "fixed",
                fixed = { 0.375, -0.5, -0.5, 0.5, 0.5, 0.5 }
        },
        node_box = {
                type = "fixed",
		fixed = {
			{  6/16, -8/16, -8/16,  8/16,  8/16, -7/16 },	-- left post
			{  6/16, -8/16,  6/16,  8/16,  8/16,  8/16 }, 	-- right post
			{ 13/32,  7/16, -8/16, 15/32,  8/16,  8/16 },	-- top piece
			{ 13/32, -8/16, -8/16, 15/32, -7/16,  8/16 },	-- bottom piece
			{  7/16, -8/16, -8/16,  7/16,  8/16,  8/16 },	-- the chainlink itself
			{  6/16, -3/16, -8/16,  8/16,  3/16, -6/16 }	-- the lump representing the lock
		}		
        },
	drop = "homedecor:fence_chainlink_gate_closed"
})

minetest.register_on_punchnode(function (pos, node)
	if node.name=="homedecor:fence_picket_gate_white_closed" then 
		fdir=node.param2
		minetest.env:add_node(pos, { name = "homedecor:fence_picket_gate_white_open", param2 = fdir })
	end
end)

minetest.register_on_punchnode(function (pos, node)
	if node.name=="homedecor:fence_picket_gate_white_open" then 
		fdir=node.param2
		minetest.env:add_node(pos, { name = "homedecor:fence_picket_gate_white_closed", param2 = fdir })
	end
end)

minetest.register_on_punchnode(function (pos, node)
	if node.name=="homedecor:fence_picket_gate_closed" then 
		fdir=node.param2
		minetest.env:add_node(pos, { name = "homedecor:fence_picket_gate_open", param2 = fdir })
	end
end)

minetest.register_on_punchnode(function (pos, node)
	if node.name=="homedecor:fence_picket_gate_open" then 
		fdir=node.param2
		minetest.env:add_node(pos, { name = "homedecor:fence_picket_gate_closed", param2 = fdir })
	end
end)

minetest.register_on_punchnode(function (pos, node)
	if node.name=="homedecor:fence_barbed_wire_gate_closed" then 
		fdir=node.param2
		minetest.env:add_node(pos, { name = "homedecor:fence_barbed_wire_gate_open", param2 = fdir })
	end
end)

minetest.register_on_punchnode(function (pos, node)
	if node.name=="homedecor:fence_barbed_wire_gate_open" then 
		fdir=node.param2
		minetest.env:add_node(pos, { name = "homedecor:fence_barbed_wire_gate_closed", param2 = fdir })
	end
end)

minetest.register_on_punchnode(function (pos, node)
	if node.name=="homedecor:fence_chainlink_gate_closed" then 
		fdir=node.param2
		minetest.env:add_node(pos, { name = "homedecor:fence_chainlink_gate_open", param2 = fdir })
	end
end)

minetest.register_on_punchnode(function (pos, node)
	if node.name=="homedecor:fence_chainlink_gate_open" then 
		fdir=node.param2
		minetest.env:add_node(pos, { name = "homedecor:fence_chainlink_gate_closed", param2 = fdir })
	end
end)


